imi/src/main/java/org/motechproject/nms/imi/service/impl/CdrFileServiceImpl.java

```diff
@@ -42,167 +59,426 @@
+                            AlertType.MEDIUM, AlertStatus.NEW, 0, null);
+                }
+            } catch (IOException e) {
+                error = String.format("Unable to send cdrFile notification request: %s", e.getMessage());
+                LOGGER.warn(error);
+                alertService.create(cfpn.getFileName(), "cdrFile notification request", error,
+                        AlertType.MEDIUM, AlertStatus.NEW, 0, null);
+            }
+            count++;
+        }
+
+        // Retry count exceeded, consider this a critical error
+        LOGGER.error(error);
+        alertService.create(cfpn.getFileName(), "cdrFile notification request", error,
+                AlertType.CRITICAL, AlertStatus.NEW, 0, null);
+
+    }
+
 
+    private void reportAuditAndPost(String file, List<String> errors) {
+        for (String error : errors) {
+            LOGGER.error(error);
+            alertService.create(file, CDR_DETAIL_FILE, error, AlertType.CRITICAL, AlertStatus.NEW, 0, null);
+            fileAuditRecordDataService.create(new FileAuditRecord(FileType.CDR_DETAIL_FILE, file, false, error,
+                    null, null));
+        }
+        sendNotificationRequest(new CdrFileProcessedNotification(
+                FileProcessedStatus.FILE_ERROR_IN_FILE_FORMAT.getValue(),
+                file,
+                StringUtils.join(errors, ",")
+        ));
+    }
+
+
+    // Aggregate detail records for a requestId into a single summary record
+    private void aggregateDetailRecord(CallDetailRecordDto cdr, CallSummaryRecordDto csr) {
+        if (csr.getRequestId() == null) {
+            csr.setRequestId(cdr.getRequestId());
+            csr.setMsisdn(cdr.getMsisdn());
+            csr.setContentFileName(cdr.getContentFile());
+            csr.setWeekId(cdr.getWeekId());
+            csr.setLanguageLocationCode(cdr.getLanguageLocationId());
+            csr.setCircle(cdr.getCircleId());
+            csr.setFinalStatus(FinalCallStatus.fromStatusCode(cdr.getStatusCode()));
+            Map<Integer, Integer> statusStats = new HashMap<>();
+            statusStats.put(cdr.getStatusCode().getValue(), 1);
+            csr.setStatusStats(statusStats);
+            csr.setSecondsPlayed(cdr.getMsgPlayDuration());
+            csr.setCallAttempts(1);
+        } else {
             // Increment the statusCode stats
             int statusCodeCount = 1;
-            if (record.getStatusStats().containsKey(cdr.getStatusCode())) {
-                statusCodeCount = record.getStatusStats().get(cdr.getStatusCode()) + 1;
+            if (csr.getStatusStats().containsKey(cdr.getStatusCode().getValue())) {
+                statusCodeCount = csr.getStatusStats().get(cdr.getStatusCode().getValue()) + 1;
             }
-            record.getStatusStats().put(cdr.getStatusCode().getValue(), statusCodeCount);
+            csr.getStatusStats().put(cdr.getStatusCode().getValue(), statusCodeCount);
 
             // Increment the message play duration
```

Comment:
It's not a duration it's a count of the number of times the called failed,
for example we tried to call this number:

2 times there was a telco line error
3 times there was no pickup
1 time they picked up and immediately disconnected
1 time they picked up and we successfully played the message


On Tue, May 26, 2015 at 5:39 PM, kosh <notifications@github.com> wrote:

> In
> imi/src/main/java/org/motechproject/nms/imi/service/impl/CdrFileServiceImpl.java
> <https://github.com/motech-implementations/mim/pull/266#discussion_r31094172>
> :
>
> >              }
> > -            record.getStatusStats().put(cdr.getStatusCode().getValue(), statusCodeCount);
> > +            csr.getStatusStats().put(cdr.getStatusCode().getValue(), statusCodeCount);
> >
> >              // Increment the message play duration
>
> I might be missing something here. if there are multiple cdr(s) for the
> same csr, why would the play duration be added?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/motech-implementations/mim/pull/266/files#r31094172>.
>



-- 
Frank Huster
Mobile Health Innovations
Grameen Foundation
+1.206.325.6690 x201